src/core/memory.d

```diff
@@ -786,35 +688,23 @@ struct GC
      *  Information regarding the memory block referenced by p or BlkInfo.init
      *  on error.
      */
-    static BlkInfo query( const scope void* p ) nothrow
-    {
-        return gc_query(cast(void*)p);
-    }
+    pragma(mangle, "gc_query") static BlkInfo query(const scope void* p) nothrow;
 
 
     /// ditto
-    static BlkInfo query(void* p) pure nothrow
-    {
-        return gc_query( p );
-    }
+    pragma(mangle, "gc_query") static BlkInfo query(void* p) pure nothrow;
 
     /**
      * Returns runtime stats for currently active GC implementation
      * See `core.memory.GC.Stats` for list of available metrics.
      */
-    static Stats stats() nothrow
-    {
-        return gc_stats();
-    }
+    pragma(mangle, "gc_stats") static Stats stats() nothrow;
```

Comment:
Remarkably/worryingly this change by itself is enough to cause the linux32 build to consistently segfault in the profilegc test. Backing out this and a few other changes in `core.memory` will probably cause the test suite to pass but I'd like to understand why this is happening.

Reply:
issue number?